ocaml/idl/datamodel_pool.ml

```diff
@@ -1024,6 +1024,46 @@ let set_uefi_certificates =
       ]
     ~allowed_roles:_R_POOL_ADMIN ()
 
+let update_sync_frequency =
+  Enum
+    ( "update_sync_frequency"
+    , [
+        ("daily", "Indicates the pool update synchronization is scheduled daily")
+      ; ( "weekly"
+        , "Indicates the pool update synchronization is scheduled weekly"
+        )
+      ; ( "monthly"
+        , "Indicates the pool update synchronization is scheduled monthly"
+        )
+      ]
+    )
+
+let configure_update_sync =
+  call ~name:"configure_update_sync"
+    ~doc:"Config scheduled job to sync update from remote CDN" ~lifecycle:[]
+    ~params:
+      [
+        (Ref _pool, "self", "The pool")
+      ; ( update_sync_frequency
+        , "update_sync_frequency"
+        , "The frequency at which updates are synced from remote CDN: daily, \
+           weekly, or monthly."
+        )
+      ; ( Int
+        , "update_sync_day"
+        , "Which day of one period the update sychronization is scheduled. For \
+           'daily' schedule, it should be 1. For 'weekly' schedule, -7..-1 and \
+           1..7, where 1 and -7 is Sunday. For 'monthly' schedule, -28..-1 and \
+           1..28, this ensure that the day exist in every month. For negative \
+           number, it means the day count down from the end of the period."
```

Comment:
For a month with 30 days, what does `-1` mean? I could read this as `30-1=29` so synchronisation happens on the 29th day. Is this how this works or is the documentation inaccurate? How would you make it run on the 30th day?

Reply:
The meaning of negative number is following python way which is suggested by Rob: -1 means the last day of that period, so for a month with 30 days, `-1` means the last day of that month: day 30.

Reply:
Python is 0-based: for an array of size 30, you get 30-1=29 which is the last entry in the array. Using this with a 1-based system is not as natural. Your explanation in the documentation "For negative number, it means the day count down from the end of the period." does not match the intended semantics.

Reply:
I think you are right! Then how about changing the `update_sync_day` to 0-based. @robhoes @minglumlu are you happy with this?

And for the comment: "For negative number, it means the day count down from the end of the period.", I am not quiet sure how to express it correctly and briefly, will this be ok:

`For negative number, it means counting down from the end of the period, which follows Python style, For example, -1 with 'monthly' schedule means the last day of the month.`

Reply:
I think Python uses `1` for the first day of a month (at least from this https://docs.python.org/3/library/datetime.html#datetime.date.day).
I don't think the `0th` day of a month looks good. Every time a user who wants to set to a day has to do a decreasing.

The start day of a week should be clarified in API/CLI document as Monday. But in this case, the number starting from either 0 or 1 can work.

I don't think we need to keep same base for all of month, week, hour. Naturally, the base day of month is 1, and the base hour of day is 0. The base day of week can be either 0 or 1.

Reply:
Thanks @minglumlu , one thing to note is that: the start day of a week is Sunday.
And as the PR is already merged, will clarify it in API/CLI document in the next PR.